apm-sniffer/apm-sdk-plugin/xxl-job-plugin/src/main/java/org/apache/skywalking/apm/plugin/xxljob/JobHandlerInterceptor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.xxljob;
+
+import com.xxl.job.core.util.ShardingUtil;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.StringTag;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+import java.lang.reflect.Method;
+
+/**
+ * @author tianjunwei
+ * 2019/4/21 10:55
+ */
+public class JobHandlerInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+
+        String triggerParam = (String) allArguments[0];
+        String operateName = null;
+        if (method != null) {
+            operateName = method.getDeclaringClass().getName();
+        } else {
+            operateName = ComponentsDefine.XXL_JOB.getName();
+        }
+        ShardingUtil.ShardingVO shardingVO = ShardingUtil.getShardingVo();
+        if (shardingVO != null) {
+
+            operateName = operateName + "_" + shardingVO.getIndex();
+        }
+        AbstractSpan span;
+        span = ContextManager.createEntrySpan(operateName, null);
```

Comment:
Create an Entry span but without ContextCarrier. Could you tell me why?

Reply:
cc @ascrutae

Reply:
Why do you create entry span? Entry span means RPC related, but I can't find another exit span somewhere. From the codes I saw, this is only a local span to group following spans.

Reply:
> I suggest implementing the cron expression job on the tracking the xxl server to deliver the task to the client.

Noticed @zhaoyuguang comments. If you have server side, you could choose to implement server side plugin, with ContextCarrier propagation. Or just create a local span, without propagation.

But you need to define and be clear, what your plugin for, and how it should work.

Reply:
the server side is web server，and the server side call client by  http （RPC），and I use the createEntrySpan

Reply:
if the server side is deployed without skywaling , the client maybe use the  createLocalSpan like elastic-job

Reply:
If this goes through HTTP, which plugin acts a web container? I don't think you handler should create entry span in either case. Entry span means the incoming call, but in the current case, I think your plugin has nothing related to RPC.

Reply:
> if the server side is deployed without skywaling , the client maybe use the createLocalSpan like elastic-job

You are creating entry span in your codes.